Services/Common/OJS.Services.Common.Models/Cache/ContestCategoryTreeViewModel.cs

```diff
@@ -22,13 +22,27 @@ public class ContestCategoryTreeViewModel : IMapExplicitly
     public IEnumerable<ContestCategoryTreeViewModel> Children { get; set; }
         = Enumerable.Empty<ContestCategoryTreeViewModel>();
 
+    public ICollection<AllowedContestStrategiesServiceModel> AllowedStrategyTypes { get; set; }
+
+
     public void RegisterMappings(IProfileExpression configuration)
         => configuration
             .CreateMap<ContestCategory, ContestCategoryTreeViewModel>()
             .ForMember(
                 m => m.Children,
                 opt => opt.MapFrom(src =>
                     src.Children
-                        .Where(c => c.IsVisible)
-                        .OrderBy(c=> c.OrderBy)));
+                    .Where(c => c.IsVisible)))
+            .ForMember(
+                m => m.AllowedStrategyTypes,
+                opt => opt.MapFrom(src =>
+                    src.Contests
+                        .Where(cc => cc.IsVisible && !cc.IsDeleted)
```

Comment:
Is it possible to remove this filtering from the mapping? Data should be ready to map before mapping it to the view model meaning that the business logic should take care of contests which are visible/not visible etc to not reach mapping if not needed.